### PR TITLE
Port of https://github.com/dotnet/coreclr/pull/12795 to release/1.1.0

### PR DESCRIPTION
**Pull request by karajas**
Display the contents of init-tools.log

/cc @ravimeda

### CODE REVIEW

File: init-tools.cmd [hunk 1]
```diff
@@ -73,10 +70,12 @@ echo Initializing BuildTools ...
...
 echo Init-Tools.cmd completed for BuildTools Version: %BUILDTOOLS_VERSION% > "%BUILD_TOOLS_SEMAPHORE%"
+exit /b 0
 
-:DONE
-
-exit /b %TOOLS_INIT_RETURN_CODE%
+:error
+echo Please check the detailed log that follows. 1>&2
+type "%INIT_TOOLS_LOG%" 1>&2
+exit /b 1
```
Comment (ravimeda) [inline init-tools.cmd]: Nit - No new line at end of file.

File: init-tools.sh [hunk 2]
```diff
@@ -110,11 +118,20 @@ if [ ! -e $__PROJECT_JSON_FILE ]; then
     echo $__PROJECT_JSON_CONTENTS > "$__PROJECT_JSON_FILE"
 
     if [ ! -e $__BUILD_TOOLS_PATH ]; then
+        echo "Running: $__DOTNET_CMD restore "$__PROJECT_JSON_FILE" --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE" >> $__init_tools_log
         $__DOTNET_CMD restore "$__PROJECT_JSON_FILE" --packages $__PACKAGES_DIR --source $__BUILDTOOLS_SOURCE
```
Comment (ravimeda) [inline init-tools.sh]: Redirect output to log. Similar to the corresponding batch file.

File: init-tools.sh [hunk 3]
```diff
@@ -110,11 +118,20 @@ if [ ! -e $__PROJECT_JSON_FILE ]; then
...
     fi
 
+    echo "Initializing BuildTools..."
+    echo "Running: $__BUILD_TOOLS_PATH/init-tools.sh $__scriptpath $__DOTNET_CMD $__TOOLRUNTIME_DIR" >> $__init_tools_log
     # On ubuntu 14.04, /bin/sh (symbolic link) calls /bin/dash by default.
     $__BUILD_TOOLS_PATH/init-tools.sh $__scriptpath $__DOTNET_CMD $__TOOLRUNTIME_DIR
```
Comment (ravimeda) [inline init-tools.sh]: Same comment about redirection applies here too.

File: init-tools.cmd (hunk 1 above)
Comment (karajas) [inline init-tools.cmd]: fixed

**Comment by karajas** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build please
@dotnet-bot test Windows_NT x64 CoreCLR Perf Tests Correctness please
@dotnet-bot test Windows_NT x86 CoreCLR Perf Tests Correctness please

**Comment by karajas** [comment]
Talked with @DrewScoggins , the failures in CoreCLR Perf are due to a bug in https://github.com/dotnet/coreclr/blob/release/1.1.0/tests/runtest.cmd#L188 will include a fix for that along with this pr and update shiproom.

**Pull request merged**